vignettes/questionnaires.Rmd

```diff
@@ -0,0 +1,627 @@
+  ) %>%
+  # add analysis parameter variables
+  mutate(
+    PARAMCD = QSTESTCD,
+    PARAM = QSTEST,
+    PARCAT1 = QSCAT,
+    AVALC = QSORRES,
+    AVAL = QSSTRESN
+  ) %>%
+  # add timing variables
+  derive_vars_dt(new_vars_prefix = "A", dtc = QSDTC) %>%
+  derive_vars_dy(reference_date = TRTSDT, source_vars = exprs(ADT)) %>%
+  mutate(
+    AVISIT = VISIT,
+    AVISITN = VISITNUM
+  )
+```
+
+```{r echo=FALSE}
+dataset_vignette(
+  arrange(adqs, USUBJID, PARCAT1, ADY, PARAMCD),
+  display_vars = exprs(USUBJID, PARAMCD, PARAM, PARCAT1, AVALC, AVAL, ADY, AVISIT)
+)
+```
+
+We handle unscheduled visits as normal visits. For deriving visits based on
+time-windows see [Visit and Period Variables](visits_periods.html#visits) and
+for flagging values to be used for analysis `derive_var_extreme_flag()`.
+
+# Scales and Scores
+
+Scales and Scores are often derived as the sum or the average across a subset of
+the items. For the GAD-7 questionnaire the total score is derived as the sum.
+The `derive_summary_records()` function with `sum()` can be used to derive it as
+a new parameter. In the example we derive a separate ADaM dataset for each
+questionnaire. Depending on the analysis needs it is also possible that an ADaM
+contains more than one questionnaire or all questionnaires.
+```{r eval=TRUE}
+adgad7 <- adqs %>%
+  filter(PARCAT1 == "GAD-7 V2") %>%
+  derive_summary_records(
+    by_vars = exprs(USUBJID, AVISIT, ADT, ADY),
+    analysis_var = AVAL,
+    summary_fun = function(x) sum(x, na.rm = TRUE),
+    filter = str_detect(PARAMCD, "GAD020[1-7]"),
+    set_values_to = exprs(
+      PARAMCD = "GAD02TS",
+      PARAM = "GAD02-Total Score - Analysis"
+    )
+  )
+```
+
+```{r echo=FALSE}
+dataset_vignette(
+  arrange(adgad7, USUBJID, ADY, PARAMCD),
+  display_vars = exprs(USUBJID, PARAMCD, PARAM, AVAL, ADY, AVISIT)
+)
+```
+
+For the GDS-SF questionnaire the total score is defined as the average of the
+item values transformed to the range [0, 15] and rounded up to the next integer.
+If more than five items are missing, the total score is considered as missing.
+This parameter can be derived by `compute_scales()` and
+`derive_summary_records()`:
+```{r eval=TRUE}
+adgdssf <- adqs %>%
+  filter(PARCAT1 == "GDS SHORT FORM") %>%
+  derive_summary_records(
+    by_vars = exprs(USUBJID, AVISIT, ADT, ADY),
+    analysis_var = AVAL,
+    summary_fun = function(x) {
+      compute_scale(
+        x,
+        source_range = c(0, 1),
+        target_range = c(0, 15),
+        min_n = 10
+      ) %>%
+        ceiling()
+    },
+    filter = str_detect(PARAMCD, "GDS02[01][0-9]"),
+    set_values_to = exprs(
+      PARAMCD = "GDS02TS",
+      PARAM = "GDS02- Total Score - Analysis"
+    )
+  )
+```
+
+```{r echo=FALSE}
+dataset_vignette(
+  arrange(adgdssf, USUBJID, ADY, PARAMCD),
+  display_vars = exprs(USUBJID, PARAMCD, PARAM, AVAL, ADY, AVISIT)
+)
+```
+
+After deriving the scores by visit baseline and change from baseline variables
+can be derived:
```

Comment:
```suggestion
After deriving the scores by visit baseline and change from baseline the variables
can be derived:
```

Reply:
Updated slightly differently. Is this OK?